File: eCommerce/eCommerce.Services/Database/RezervacijaProstora.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCommerce.Services.Database
{
    public class RezervacijaProstora
    {
        public int RezervacijaProstoraId { get; set; }
        public int RadniProstorId { get; set; }
        public RadniProstor RadniProstor { get; set; }
        public int KorisnikId { get; set; }
        public User Korisnik { get; set; }
        public DateTime DatumEvidencije { get; set; }
        public int Trajanje { get; set; }
        public Status StatusRezervacije { get; set; }
        public string Napomena { get; set; }
        public enum Status
        {
            Aktivna,
            Otkazana,
            Potvrdjena
        }
    }
}

File: eCommerce/UI/ecommerce_mobile/lib/model/rezervacijaprostora.dart
import 'package:ecommerce_mobile/model/radniprostor.dart';
import 'package:ecommerce_mobile/model/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rezervacijaprostora.g.dart';

@JsonSerializable()
class RezervacijaProstora {
  int? rezervacijaProstoraId;
  int? radniProstorId;
  int? korisnikId;
  User? korisnik;
  RadniProstor? radniProstor;
  DateTime? datumRezervacije;
  int? trajanje;
  String? status;
  String? napomena;

  RezervacijaProstora({
    this.rezervacijaProstoraId,
    this.radniProstorId,
    this.korisnikId,
    this.korisnik,
    this.radniProstor,
    this.datumRezervacije,
    this.trajanje,
    this.status,
    this.napomena,
  });

  factory RezervacijaProstora.fromJson(Map<String, dynamic> json) => _$RezervacijaProstoraFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijaProstoraToJson(this);
} 